bids_conversion/convert_somato_data.py

```diff
@@ -0,0 +1,268 @@
+"""Convert the MNE-somato-data to BIDS format.
+
+Requirements
+------------
+The python dependencies to run this script are:
+
+- numpy>=1.16
+- scipy>=1.2
+- mne<0.19
+- mne-bids>=0.3
+- nibabel>=2.4.1
+
+freesurfer
+----------
+Next to the python requirements, you will need freesurfer installed on your
+system. Once freesurfer is installed and you want to run this script, make sure
+to initialize freesurfer first:
+
+$ export FREESURFER_HOME=/path/to/FreeSurfer
+$ source $FREESURFER_HOME/SetUpFreeSurfer.sh
+
+and remember to define a SUBJECTS_DIR, where freesurfer will copy the data to:
+
+$ export SUBJECTS_DIR=/path/to/subjects
+
+Helpful links
+-------------
+- https://mne-tools.github.io/stable/index.html
+- https://mne-tools.github.io/mne-bids/
+- https://mne-tools.github.io/stable/manual/datasets_index.html?#somatosensory
+- https://surfer.nmr.mgh.harvard.edu/fswiki/FreeSurferWiki
+- https://mne-tools.github.io/stable/auto_tutorials/source-modeling/plot_background_freesurfer.html  # noqa: E501
+
+"""
+# Authors: Stefan Appelhoff <stefan.appelhoff@mailbox.org>
+#
+# License: BSD (3-clause)
+import os
+import os.path as op
+import shutil as sh
+import json
+import warnings
+
+import numpy as np
+import mne
+from mne.utils import check_version, run_subprocess
+import mne.datasets.somato as somato
+from mne_bids import write_raw_bids, make_bids_basename, write_anat
+from mne_bids.utils import print_dir_tree
+
+# mne should be 0.18 minimum ... but not higher
+if not check_version('mne', '0.18') or check_version('mne', '0.19'):
+    raise RuntimeError('You need to install mne 0.18: `pip install mne==0.18`')
+# the other dependencies are a bit more lenient, we check anyways
+for pkg, ver in [('numpy', '1.16'), ('scipy', '1.2'), ('nibabel', '2.4.1'),
+                 ('mne_bids', '0.3')]:
```

Comment:
This *should* work, but it doesn't because the `dev` version in MNE-BIDS was set wrong: https://github.com/mne-tools/mne-bids/issues/204

If you want to run this script now, either fix mne_bids's dev version ... or change the requirement to `0.2` (but still use master of mne-bids)

Reply:
This check is a bit whiny but it's fine. Hopefully it should be okay after a few months.

Reply:
our milestone in mne-bids is actually set for today: https://github.com/mne-tools/mne-bids/milestone/3

:-)

I would release `0.3` before my GSoC is over, there's been a lot of cool new stuff, bug fixes, and better docs - no need to keep the userbase (which is slowly growing now) waiting